libs/model-assessment/src/lib/ModelAssessmentDashboard/ModelAssessmentDashboard.tsx

```diff
@@ -968,4 +488,34 @@ export class ModelAssessmentDashboard extends React.PureComponent<
     );
     this.setState({ selectedWeightVector: weightOption });
   };
+
+  private handleErrorDetectorChanged = (
+    _: React.FormEvent<HTMLDivElement>,
+    item?: IDropdownOption
+  ): void => {
+    if (item) {
+      if (item.key === ErrorAnalysisOptions.HeatMap) {
+        // Note comparison above is actually string comparison (key is string), we have to set the enum
+        const selectedOptionHeatMap = ErrorAnalysisOptions.HeatMap;
+        this.setErrorDetector(selectedOptionHeatMap);
+      } else {
+        // Note comparison above is actually string comparison (key is string), we have to set the enum
+        const selectedOptionTreeMap = ErrorAnalysisOptions.TreeMap;
+        this.setErrorDetector(selectedOptionTreeMap);
+      }
+    }
+  };
+
+  private setErrorDetector = (key: ErrorAnalysisOptions): void => {
+    if (this.state.selectedCohort.isTemporary) {
+      this.setState({
+        mapShiftErrorAnalysisOption: key,
+        openMapShift: true
+      });
+    } else {
+      this.setState({
+        errorAnalysisOption: key
+      });
+    }
+  };
```

Comment:
And there I was thinking 500 is great :-) It was at 1300 before I starting moving things into their own components. I'd probably prefer doing it in a separate PR since this one already got so large from my first attempts at moving things into their own components. The use of the ModelAssessmentContext should help me trim it more as well.